File: backend/routes/properties.py
# =========================
# Properties Routes
# =========================

from flask import Blueprint, request, jsonify
from models import Property
from extensions import db

properties_bp = Blueprint('properties', __name__)


# =========================
# GET all properties
# =========================
@properties_bp.route('/properties', methods=['GET'])
def get_properties():

    properties = Property.query.all()

    result = []

    for prop in properties:
        result.append({
            "id": prop.id,
            "name": prop.name,
            "address": prop.address,
            "notes": prop.notes,
            "created_at": prop.created_at
        })

    return jsonify(result), 200


# =========================
# GET single property
# =========================
@properties_bp.route('/properties/<int:id>', methods=['GET'])
def get_property(id):

    prop = Property.query.get_or_404(id)

    return jsonify({
        "id": prop.id,
        "name": prop.name,
        "address": prop.address,
        "notes": prop.notes,
        "created_at": prop.created_at
    }), 200


# =========================
# CREATE property
# =========================
@properties_bp.route('/properties', methods=['POST'])
def create_property():

    data = request.get_json()

    if not data or not data.get('name'):
        return jsonify({"error": "Property name is required"}), 400

    new_property = Property(
        name=data['name'],
        address=data.get('address'),
        notes=data.get('notes'),
        user_id=1  # temporary user for MVP
    )

    db.session.add(new_property)
    db.session.commit()

    return jsonify({
        "message": "Property created successfully",
        "id": new_property.id
    }), 201


# =========================
# DELETE property
# =========================
@properties_bp.route('/properties/<int:id>', methods=['DELETE'])
def delete_property(id):

    prop = Property.query.get_or_404(id)

    db.session.delete(prop)
    db.session.commit()

    return jsonify({"message": "Property deleted"}), 200

File: backend/routes/properties.js
const express = require('express');
const router = express.Router();
const auth = require('../middleware/auth');
const db = require('../db');
router.use(auth);
router.get('/', async (req, res) => {
  try {
    const r = await db.query('SELECT * FROM properties WHERE user_id = $1 ORDER BY created_at DESC', [req.userId]);
    res.json(r.rows);
  } catch (err) { res.status(500).json({ error: 'Failed' }); }
});
router.post('/', async (req, res) => {
  try {
    const { name, address, type, beds, status, emoji, notes, tags } = req.body;
    if (!name) return res.status(400).json({ error: 'Name required' });
    const r = await db.query('INSERT INTO properties (name, address, type, beds, status, emoji, notes, tags, user_id) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9) RETURNING *', [name, address||'', type||'Airbnb', beds||1, status||'active', emoji||'🏠', notes||'', JSON.stringify(tags||[]), req.userId]);
    res.status(201).json(r.rows[0]);
  } catch (err) { console.error(err); res.status(500).json({ error: 'Failed' }); }
});
router.delete('/:id', async (req, res) => {
  try {
    await db.query('DELETE FROM properties WHERE id = $1 AND user_id = $2', [req.params.id, req.userId]);
    res.json({ message: 'Deleted' });
  } catch (err) { res.status(500).json({ error: 'Failed' }); }
});
module.exports = router;